Queues/FileCommandQueueHandler.cs

```diff
@@ -80,25 +91,26 @@ public IEnumerable<ICommand> GetCommands()
         }
 
         /// <summary>
-        /// Reads the commnd file on disk and populates the queue with the commands
+        /// Initializes this command queue for the specified job
         /// </summary>
-        private void ReadCommandFile()
+        /// <param name="job">The job that defines what queue to bind to</param>
+        /// <param name="algorithm">The algorithm instance</param>
+        public void Initialize(AlgorithmNodePacket job, IAlgorithm algorithm)
         {
-            object deserialized;
-            try
-            {
-                if (!File.Exists(_commandJsonFilePath)) return;
-                var contents = File.ReadAllText(_commandJsonFilePath);
-                deserialized = JsonConvert.DeserializeObject(contents, new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.All });
-            }
-            catch (Exception err)
-            {
-                Log.Error(err);
-                deserialized = null;
-            }
+        }
 
-            // remove the file when we're done reading it
-            File.Delete(_commandJsonFilePath);
+        /// <summary>
+        /// Reads the command file on disk and deserialize to object
+        /// </summary>
+        protected abstract object ReadCommandFile();
```

Comment:
I think ideally this would return `IEnumerable<ICommand>` -- returning object is a vague contract.